imsim/readout.py

```diff
@@ -239,13 +252,32 @@ def get_primary_hdu(eimage, lsst_num, camera_name=None,
     dectel = eimage.header['DECTEL']
     rottelpos = eimage.header['ROTTELPOS']
     band = eimage.header['FILTER']
+    # Compute rotSkyPos instead of using likely inconsistent values
+    # from the instance catalog or opsim db.
+    mjd_obs = eimage.header['MJD-OBS']
+    mjd_end =  mjd_obs + exptime/86400.
+    rotang = compute_rotSkyPos(
+        ratel, dectel, rottelpos, mjd_obs, band, camera_name=camera_name,
+        logger=logger
+    )
+    phdu.header['ROTANGLE'] = rotang
     if camera_name == 'LsstCamImSim':
         phdu.header['FILTER'] = band
         phdu.header['TESTTYPE'] = 'IMSIM'
         phdu.header['RAFTNAME'] = raft
         phdu.header['SENSNAME'] = sensor
         phdu.header['RATEL'] = ratel
         phdu.header['DECTEL'] = dectel
+    elif camera_name == 'LsstComCam':
+        phdu.header['FILTER'] = ComCam_filter_map.get(band, None)
+        phdu.header['TELESCOP'] = SIMONYI_TELESCOPE
+        phdu.header['INSTRUME'] = 'ComCam'
```

Comment:
Is this the right value for `INSTRUME`?  Based on the convention for LSSTCam and the existing datasets in `/repo/main` at S3DF:
```
[lsstccs@sdfrome001 LSSTCam w_2023_49] butler query-collections /repo/main */raw/all
       Name        Type
------------------ ----
DECam/raw/all      RUN 
HSC/raw/all        RUN 
LATISS/raw/all     RUN 
LSST-TS8/raw/all   RUN 
LSSTCam/raw/all    RUN 
LSSTComCam/raw/all RUN 
```
I would have expected the desired value to be `LSSTComCam`.

Reply:
I checked an existing ComCam raw file in /repo/main, and it does indeed use `ComCam`, so this is ok.